test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/ESClientYamlSuiteTestCase.java

```diff
@@ -220,7 +220,26 @@ public static Iterable<Object[]> createParameters() throws Exception {
      * Create parameters for this parameterized test.
      */
     public static Iterable<Object[]> createParameters(NamedXContentRegistry executeableSectionRegistry) throws Exception {
-        String[] paths = resolvePathsProperty(REST_TESTS_SUITE, ""); // default to all tests under the test root
+        return createParameters(executeableSectionRegistry, null);
+    }
+
+    /**
+     * Create parameters for this parameterized test.
+     */
+    public static Iterable<Object[]> createParameters(String[] testPaths) throws Exception {
+        return createParameters(ExecutableSection.XCONTENT_REGISTRY, testPaths);
+    }
+
+    /**
+     * Create parameters for this parameterized test.
+     *
+     * @param executeableSectionRegistry registry of executable sections
+     * @param testPaths list of paths to explicitly search for tests. If <code>null</code> then include all tests in root path.
+     * @return list of test candidates.
+     * @throws Exception
+     */
+    public static Iterable<Object[]> createParameters(NamedXContentRegistry executeableSectionRegistry, String[] testPaths) throws Exception {
+        String[] paths = testPaths == null ? resolvePathsProperty(REST_TESTS_SUITE, "") : testPaths; // default to all tests under the test root
```

Comment:
Since the REST_TESTS_SUITE parameter is used for repro lines, wouldn't any testPaths passed in need to be filtered against it?

Reply:
Good question, these are really designed to be mutually exclusive. Perhaps we should just error if you attempt to set that system property _and_ pass an explicit paths property?

Reply:
Realistically if you are trying to run a specific yaml test you use the repro line that includes the explicit class and test name system properties.

Reply:
The test will always pass the explicit paths right? If a reproduce line is passed, we would have both, so that shouldn't be an error. But I guess in this case the repro line has the single test we want. The concern I have is if the property is not filtered, every suite will try to execute the system property based test, when there will only be one that it is applicable to. So if the the system property is filtered against the explicit paths, suites that are not applicable would return an empty list and not execute.

Reply:
The `tests.rest.suite` property would only be included in the repro line if it was explicitly passed as a system property in the first place, which it is only done so in very specific tests suites like upgrade tests that have "phases". My thought is if you use the `tests.rest.suite` system property to filter tests, you shouldn't pass the paths programmatically via this new mechanism and vice versa and so perhaps we should explicitly disallow it so there's no ambiguity as to which would take precedence over the other.